### PR TITLE
[IMP] web: display and can delete default_group_by

### PR DESCRIPTION
**Pull request by juliusc2066**
This commit improves the default_group_by UX by displaying it in the search facets and allowing the user to remove it when needed.

task-3895650

### CODE REVIEW

File: addons/web/static/src/views/kanban/progress_bar_hook.js [hunk 1]
```diff
@@ -246,16 +246,15 @@ class ProgressBarState {
...
+        if (groupBy.length) {
             const resModel = this.model.root.resModel;
             const domain = this.model.root.domain;
             const context = this.model.root.context;
             const { colors, fieldName: field, help } = this.progressAttributes;
             const groupsId = this.model.root.groups.map((g) => g.id).join();
             const res = await this.model.orm.call(resModel, "read_progress_bar", [], {
                 domain,
-                group_by: groupBy.length ? groupBy[0] : defaultGroupBy,
+                group_by: groupBy.length ? groupBy[0] : [],
```
Comment (Polymorphe57) [inline addons/web/static/src/views/kanban/progress_bar_hook.js:257]:
```suggestion
                group_by: groupBy[0],
```

**Comment by Polymorphe57** [comment]
add tests

File: addons/web/static/src/search/search_model.js [hunk 2]
```diff
@@ -1748,7 +1772,11 @@ export class SearchModel extends EventBus {
                 }
             }
         }
-        const groupBy = groupBys.length ? groupBys : this.globalGroupBy.slice();
+        const groupBy = groupBys.length
+            ? groupBys
+            : this.globalGroupBy.length
+            ? this.globalGroupBy.slice()
+            : this.defaultGroupBy || [];
```
Comment (Polymorphe57) [inline addons/web/static/src/search/search_model.js:1779]: a slice would be good too

File: addons/web/static/src/search/search_model.js [hunk 3]
```diff
@@ -1648,7 +1655,24 @@ export class SearchModel extends EventBus {
...
+        const hasAGroupByFacet = facets.some((f) => f.type === "groupBy");
+        if (!hasAGroupByFacet && !this.globalGroupBy.length && this.defaultGroupBy) {
+            facets.push({
+                groupId: SPECIAL,
+                type: "groupBy",
+                values: this.defaultGroupBy.map((gb) => {
+                    const [fieldName, interval] = gb.split(":");
+                    const { string } = this.searchViewFields[fieldName];
+                    if (interval) {
+                        return `${string}:${interval}`;
```
Comment (Polymorphe57) [inline addons/web/static/src/search/search_model.js:1667]: translate interval?

File: addons/web/static/tests/views/list/list_view.test.js [hunk 4]
```diff
@@ -14915,17 +14915,61 @@ test(`list view with default_group_by`, async () => {
...
+        type: "list",
+        arch: `
+            <list default_group_by="foo,bar">
+                <field name="bar"/>
+            </list>
+        `,
+    });
+    expect(`.o_list_renderer table`).toHaveClass("o_list_table_grouped");
+    expect(`.o_group_header`).toHaveCount(3);
+    if (queryAll(".o_control_panel_navigation > button").length) {
```
Comment (Polymorphe57) [inline addons/web/static/tests/views/list/list_view.test.js:14962]: why this?

Comment (juliusc2066) [reply addons/web/static/tests/views/list/list_view.test.js:14962]: to fit with mobile case without duplicating the test

**Comment by Polymorphe57** [comment]
@robodoo r+

**Pull request closed without merging**